src/main/groovy/com/calgaryscientific/gradle/VeracodeUpdateMitigationInfoTask.groovy

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * MIT License
+ *
+ * Copyright (c) 2017 Calgary Scientific Incorporated
+ *
+ * Copyright (c) 2013-2014 kctang
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ ******************************************************************************/
+
+package com.calgaryscientific.gradle
+
+import groovy.transform.CompileStatic
+
+@CompileStatic
+class VeracodeUpdateMitigationInfoTask extends VeracodeTask {
+    static final String NAME = 'veracodeUpdateMitigationInfo'
+    String build_id
+    String flaw_id_list
+    String action
+    String comment
+
+    VeracodeUpdateMitigationInfoTask() {
+        description = "Updates flaw information for the given 'build_id' and 'flaw_id_list'"
+        requiredArguments << 'build_id' << 'flaw_id_list' << 'action' << 'comment'
+        build_id = project.findProperty("build_id")
+        flaw_id_list = project.findProperty("flaw_id_list")
+        action = project.findProperty("action")
+        comment = project.findProperty("comment")
+    }
+
+    File getOutputFile() {
+        String file_flaw_id_list = flaw_id_list.replaceAll(',', '_')
+        return new File("${project.buildDir}/veracode", "mitigationinfo-${build_id}-${file_flaw_id_list}.xml")
+    }
+
+    void run() {
```

Comment:
do we have any tests that verify if the action is invalid, some exception would be raised? and I'm pretty sure that it would be hard to write this test ... because validAction is static function :) :) :)

Reply:
Test for validAction:

https://github.com/calgaryscientific/veracode-gradle-plugin/blob/385d805e859288e06db801288feb888c8eab544b/src/test/groovy/com/calgaryscientific/gradle/VeracodeMitigationInfoTest.groovy#L32

Test for validComment:

https://github.com/calgaryscientific/veracode-gradle-plugin/blob/385d805e859288e06db801288feb888c8eab544b/src/test/groovy/com/calgaryscientific/gradle/VeracodeMitigationInfoTest.groovy#L44

Reply:
hahahahah ... yes I saw those tests ... but one of the behavior, you would like to test is this .... if the comment is invalid, you want to know if an exception (fail) has been raised ...... how are you going to implement this test?

Reply:
Take a look at: https://github.com/calgaryscientific/veracode-gradle-plugin/pull/33/commits/950967e45327b08d2a0b2d9c772f1c8295fbef78